units/units.cpp

```diff
@@ -3729,26 +3758,11 @@ static bool cleanUnitString(std::string& unit_string, std::uint32_t match_flags)
         if (ReplaceStringInPlace(unit_string, " per ", 5, "/", 1)) {
             skipMultiply = true;
         }
-        auto fndP = unit_string.find(" s");
-        while (fndP != std::string::npos) {
-            if (fndP + 2 == unit_string.size()) {
-                unit_string[fndP] = '*';
-            } else {
-                switch (unit_string[fndP + 2]) {
-                    case ' ':
-                    case '*':
-                    case '/':
-                    case '^':
-                    case '.':
-                        unit_string[fndP] = '*';
-                        break;
-                    default:
-                        break;
-                }
-            }
-            fndP = unit_string.find(" s", fndP + 1);
-        }
-        fndP = unit_string.find(" of ");
+        checkShortUnits(unit_string, " s", " second");
+        checkShortUnits(unit_string, " m", " meter");
+        checkShortUnits(unit_string, " l", " liter");
```

Comment:
I don't know the bigger context of this code, but I wonder why this does not include other single-character units such as `A` or `g`?

Reply:
The specific issue is how spaces are handled.  "kg m"  should rightfully be interpreted as "kg*m"  so but "sq m"  needs to be interpreted as square meter so it has to be a bit more intelligent.  So it would be wrong to simply convert spaces to multiplies in all cases.    The particular issue in this case was that `kg m` in this case with the space removal  could also be interpreted as "kilo gm"  which happens to be kilogram with alternate abbreviations.    There were a few other cases where something similar is true.    it basically comes down to two letter units starting with a SI exponent followed by a 1 letter unit where the second letter of the first unit and the single letter of the second unit also form a valid two letter unit abbreviation.    

I should do a more thorough investigation of such just to be sure but on making the fix none others came up as obvious.  
I will investigate more, there is probably others but also probably rather uncommon in actual practice.

I did note in fixing this particular issue that the previous method would have failed for for 'sq s' or 'cu l'  which probably aren't that common but still should have been recognized.